File: src/plugin/inter/EmoteInterfacePlugin.java
package plugin.inter;

import org.redrune.game.entity.actor.player.Player;
import org.redrune.game.content.plugin.type.InterfacePlugin;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 8/31/2017
 */
public class EmoteInterfacePlugin extends InterfacePlugin {
	
	@Override
	public boolean handle(Player player, int interfaceId, int componentId, int itemId, int slotId, int packetId) {
		player.getEmotesManager().useBookEmote(componentId);
		return true;
	}
	
	@Override
	public void register() {
		registerInterfacePlugin(464);
	}
}

File: src/main/java/plugin/rsinterface/EmoteInterfacePlugin.kt
package plugin.rsinterface

import game.content.plugin.type.InterfacePlugin
import game.entity.actor.player.Player

/**
 * @author Tyluur <itstyluur></itstyluur>@icloud.com>
 * @since 8/31/2017
 */
class EmoteInterfacePlugin : InterfacePlugin {
    override fun handle(
        player: Player,
        interfaceId: Int,
        componentId: Int,
        itemId: Int,
        slotId: Int,
        packetId: Int
    ): Boolean {
        player.emotesManager.useBookEmote(componentId)
        return true
    }

    override fun register() {
        registerInterfacePlugin(464)
    }
}